python/xorbits/xgboost/mars_adapters/core.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+try:
+    import xgboost
+except ImportError:  # pragma: no cover
+    xgboost = None
+
+MARS_XGBOOST_CALLABLES = {}
+
+if xgboost is not None:
+    import functools
+    import inspect
+    from collections.abc import Iterable
+    from typing import Callable, Dict, List, Optional
+
+    from ..._mars.learn.contrib.xgboost.classifier import (
+        XGBClassifier as MarsXGBClassifier,
+    )
+    from ..._mars.learn.contrib.xgboost.regressor import (
+        XGBRegressor as MarsXGBRegressor,
+    )
+    from ...core.adapter import mars_xgboost, wrap_mars_callable
+
+    class BaseXGB:
+        def __init__(self, *args, **kwargs):
+            self.mars_instance = self.mars_cls(*args, **kwargs)
+
+        def __dir__(self) -> Iterable[str]:
+            return self.mars_instance.__dir__()
+
+    class XGBClassifier(BaseXGB):
+        mars_cls = MarsXGBClassifier
+        __doc__ = MarsXGBClassifier.__doc__
+
+    class XGBRegressor(BaseXGB):
+        mars_cls = MarsXGBRegressor
+        __doc__ = MarsXGBClassifier.__doc__
+
+    xgboost_class_mappings: Dict = {
+        XGBClassifier: MarsXGBClassifier,
+        XGBRegressor: MarsXGBRegressor,
+    }
+
+    def wrap_cls_func(mars_cls: Callable, name: str):
+        @functools.wraps(getattr(mars_cls, name))
+        def wrapped(self, *args, **kwargs):
+            return getattr(self.mars_instance, name)(*args, **kwargs)
+
+        return wrap_mars_callable(
+            wrapped,
+            attach_docstring=True,
+            is_cls_member=False,
+            docstring_src_module=xgboost,
+            docstring_src=mars_cls,
+        )
+
+    def _collect_module_callables(
+        skip_members: Optional[List[str]] = None,
+    ) -> Dict[str, Callable]:
+        module_callables: Dict[str, Callable] = dict()
+
+        for k, v in xgboost_class_mappings.items():
```

Comment:
I think it would be better to install class methods in an individual function. Since you named this function `_collect_module_callables`, it should handle only the module callables.

Reply:
pandas use this func to install cls method as well
```python
# functions and class constructors defined by mars dataframe
def _collect_module_callables() -> Dict[str, Callable]:
    mars_dataframe_callables = dict()

    # install class constructors.
    mars_dataframe_callables[mars_dataframe.DataFrame.__name__] = wrap_mars_callable(
        mars_dataframe.DataFrame,
        attach_docstring=True,
        is_cls_member=False,
        docstring_src_module=pandas,
        docstring_src=pandas.DataFrame,
    )
    mars_dataframe_callables[mars_dataframe.Series.__name__] = wrap_mars_callable(
        mars_dataframe.Series,
        attach_docstring=True,
        is_cls_member=False,
        docstring_src_module=pandas,
        docstring_src=pandas.Series,
    )
    mars_dataframe_callables[mars_dataframe.Index.__name__] = wrap_mars_callable(
        mars_dataframe.Index,
        attach_docstring=True,
        is_cls_member=False,
        docstring_src_module=pandas,
        docstring_src=pandas.Index,
    )
```